plugins/obs-ffmpeg/obs-ffmpeg.c

```diff
@@ -165,6 +167,59 @@ static bool nvenc_device_available(void)
 extern bool load_nvenc_lib(void);
 #endif
 
+#ifdef __linux__
+static int get_id_from_sys(char *d_name, char *type)
+{
+	char file_name[1024];
+	char *c;
+	int id;
+
+	snprintf(file_name, sizeof(file_name), "/sys/bus/pci/devices/%s/%s",
+		 d_name, type);
+	if ((c = os_quick_read_utf8_file(file_name)) == NULL) {
+		return -1;
+	}
+	id = strtol(c, NULL, 16);
+	bfree(c);
+
+	return id;
+}
+
+static bool nvenc_device_available(void)
+{
+	os_dir_t *dir;
+	struct os_dirent *dirent;
+	bool available = false;
```

Comment:
nit: available isnt needed. Return true instead of breaking the loop and return false at the end of the function.